src/ftof.cpp

```diff
@@ -0,0 +1,59 @@
+/*! \file ftof.cpp
+ *
+ * \brief Contains functions to transfer EigenMatrix and EigenVector to FASP
+ *  or to tranfer FASP to Fenics
+ */
+
+#include "ftof.h"
+
+using namespace dolfin;
+
+// Link the dolfin::EigenMatrix Mat_A to dCSRmat format from FASP
+void EigenMatrixTOdCSRmat( dCSRmat* dCSR_A, const EigenMatrix* mat_A)
```

Comment:
#1 : I am not using the boost library. boost::tuple does not work for me.

#2: I am agree with you

#3: just personal preference when it's a pointer I like to put it in
arguments but I really don't mind to change it as a function. Up to you.

On Sun, Oct 4, 2015 at 1:12 AM, Maximilian Metti <notifications@github.com>
wrote:

> In src/ftof.cpp
> <https://github.com/thepnpsolver/modular-pnp/pull/3#discussion_r41094506>:
>
> > @@ -0,0 +1,59 @@
> > +/*! \file ftof.cpp
> > + *
> > + * \brief Contains functions to transfer EigenMatrix and EigenVector to FASP
> > + *  or to tranfer FASP to Fenics
> > + */
> > +
> > +#include "ftof.h"
> > +
> > +using namespace dolfin;
> > +
> > +// Link the dolfin::EigenMatrix Mat_A to dCSRmat format from FASP
> > +void EigenMatrixTOdCSRmat( dCSRmat* dCSR_A, const EigenMatrix* mat_A)
>
> I'm not sure if one way is better than the other, but is there any reason
> why you chose to pass a pointer to the dCSRmat rather than return it?
> This could make test_ftof.cpp easier to read.
>
> dCSRmat EigenMatrixTOdCSRmat(const EigenMatrix* mat_A) {
>   dCSRmat dCSR_A;
>   // do some magic
>   return dCSR_A;
> }
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/thepnpsolver/modular-pnp/pull/3/files#r41094506>.
>